### DISCUSSION TITLE
Run GitHub Actions for specific number of approvals

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#26878] [Actions]
<p>Hello everyone!</p>
<p>I want to run a step in a job for a specific number of approvals on a PR. (please check the comment in below mentioned code)</p>
<p>Here is my workflow yaml:</p>
<pre><code class="lang-yaml">name: MyWorkflow
on:
  issue_comment:
    types: [created]
    branches:
      - master

jobs:
     
  merge:  
    if: contains(github.event.comment.html_url, '/pull/') &amp;&amp; startsWith(github.event.comment.body, '/merge')
    runs-on: ubuntu-latest
    steps:
      - name: Add a merge label 
        uses: actions-ecosystem/action-add-labels@v1
        with:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          labels: merge

## OR the other way is: in above step where we add label we can add label when specific number of people approved the PR.
      
      - name: automerge      
        uses: "pascalgn/automerge-action@f81beb99aef41bb55ad072857d43073fba833a98"
        env:          
          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
          MERGE_LABELS: "merge,!WIP"
          MERGE_METHOD: "squash"
          MERGE_FORKS: "false"
          MERGE_RETRIES: "6"
          MERGE_RETRY_SLEEP: "10000"
          UPDATE_LABELS: ""
          UPDATE_METHOD: "rebase" 
</code></pre>
<p>Can anyone help me to generate this  specific condition.</p>
<p>Q.2 And it will merge for the event <code>/merge</code> in the issue_comment can we restrict it to the role of the user who is commenting. Like if he’s the author or maintainer.</p>
<p>Thanks !</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/uditgaurav">@uditgaurav</a> ,</p>
<p>According to the introduction from the README of the action “<a href="https://github.com/pascalgn/automerge-action#automerge-action" rel="noopener nofollow ugc">pascalgn/automerge-action</a>”, the action will automatically check if the required number of review approvals has been reached. If the number is not reached, it will not merge the PR.</p>
<p>In addition, the event you set for the workflow is not the right. You should set the events that are related to pull request, such as “<a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc"><strong>pull_request</strong></a>”, “<a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review" rel="noopener nofollow ugc"><strong>pull_request_review</strong></a>”, “<a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review_comment" rel="noopener nofollow ugc"><strong>pull_request_review_comment</strong></a>” or “<a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target" rel="noopener nofollow ugc"><strong>pull_request_target</strong></a>”.<br>
In your case, the “<strong>pull_request_review</strong>” event should be the suitable.</p>
<p>About the second question, you can try to set the “<a href="https://docs.github.com/en/github/administering-a-repository/about-protected-branches" rel="noopener nofollow ugc"><strong>Branch protection rules</strong></a>” for target branch of the PR.</p>

**Comment by uditgaurav** [comment]
<p>Awesome thanks <a class="mention" href="/brightran">@brightran</a></p>

**Comment by BrightRan** [comment]
<p>You’re welcome, <a class="mention" href="/uditgaurav">@uditgaurav</a>.<br>
If you have any other questions about this topic, feel free to contact us.</p>
